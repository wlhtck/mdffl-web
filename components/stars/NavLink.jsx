import React from 'react';
import styled from 'styled-components';
import { string } from 'prop-types';
import { white } from '../nebulas/colors';

const NavLink = ({ url, text, className }) => (
  <a href={url} className={className}>
    {text}
  </a>
);

NavLink.propTypes = {
  url: string.isRequired,
  text: string.isRequired,
  className: string,
};

NavLink.defaultProps = {
  className: '',
};

export default styled(NavLink)`
  color: ${white};
  font-family: 'Raleway', sans-serif;
  font-size: 16px;
  text-decoration: none;
`;
