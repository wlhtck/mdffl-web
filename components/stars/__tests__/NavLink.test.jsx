import React from 'react';
import 'jest-styled-components';
import { mount } from 'enzyme';
import NavLink from '../NavLink';

describe('components/starts/NavLink', () => {
  it('renders a link', () => {
    expect(mount(<NavLink text="Button" url="/about" />)).toMatchSnapshot();
  });
});
