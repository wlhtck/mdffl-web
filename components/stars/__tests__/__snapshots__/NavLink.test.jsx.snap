// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/starts/NavLink renders a link 1`] = `
.c0 {
  color: #FFFFFF;
  font-family: 'Raleway',sans-serif;
  font-size: 16px;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<NavLink
  text="Button"
  url="/about"
>
  <NavLink
    className="c0"
    text="Button"
    url="/about"
  >
    <a
      className="c0"
      href="/about"
    >
      Button
    </a>
  </NavLink>
</NavLink>
`;
